### DISCUSSION TITLE
Ledring on another pin

### DISCUSSION THREAD

**Discussion by h4writer** [bitcraze/discussions#1093] [Q&A]
Dear,

by default IO2 / PB5 is used for communication with the ledring.
Now I would like to use IO4 / PC12. Wiring has been the easy job, but now I need to adjust the firmware.
I've looked at the ws2812_cf2.c code and I don't understand everything, but I've altered everything I think is needed.

GPIO_Pin_5 -> GPIO_Pin_12
GPIO_PinSource5 -> GPIO_PinSource12
GPIOB -> GPIOC

I've also tried SDA / PB7, and IO3 / PB4, but I get none working.

So I'm starting to think it is not possible to run a ws2812 from another pin, maybe?
Would it be possible to run the ledring from PC12/ PB7 or PB4?

Thanks Hannes

**Comment by tobbeanton** [comment]
The WS2812 driver requires a timer with DMA and might be a bit complicated to move. It is currently on TIM3_CH2 and DMA1_Stream5. The pin you want to move it to needs a free timer and a corresponding free DMA channel.
